### ISSUE TITLE
Nested record projection should generate ROW constructor keyword for rows of degree 1 in PostgreSQL's RETURNING clauses

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13119]
Some nested record projections don't work correctly where natively supported if the row has degree 1, e.g. in PostgreSQL, these aren't the same:

```sql
-- Not a nested record
SELECT (1)

-- A nested record
SELECT ROW (1)
```

We already correctly render this in `RowAsField`, but that rendering logic doesn't get triggered when using `ROW` in `RETURNING` clauses:

```sql
INSERT INTO t (a, b)
VALUES (1, 2)
-- These are different:
RETURNING (c), ROW (c)
```

Related:

- https://github.com/jOOQ/jOOQ/issues/4727

**Comment by lukaseder** [comment]
Probably not the most pressing bug, people probably don't use nested `ROW` expressions in `RETURNING` too often, let alone a single-element `ROW()`

**Comment by lukaseder** [comment]
The simplest fix is to *always* render the `ROW` keyword, irrespective of degree or context, just as we do for Informix.

**Comment by lukaseder** [comment]
Alternatively, just as for #4727, we could override the  `SelectFieldList::acceptElement` method and render `RowAsField` instead of `Row` directly.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0.

Given how edge-casey this problem is, I'm not backporting the fix.
